rdr_service/api/nph_participant_biobank_order_api.py

```diff
@@ -0,0 +1,72 @@
+import json
+from flask import request
+import logging
+from werkzeug.exceptions import BadRequest, NotFound
+
+from rdr_service.api.base_api import UpdatableApi
+from rdr_service.dao import database_factory
+from rdr_service.dao.study_nph_dao import NphOrderDao
+
+
+class CustomEncoder(json.JSONEncoder):
+    def default(self, o):
+        return o.__dict__
+
+
+def construct_response(order):
+    # Construct Response payload
+    return json.loads(json.dumps(order, indent=4, cls=CustomEncoder))
+
+
+class NphOrderApi(UpdatableApi):
+
+    def __init__(self):
+        super(NphOrderApi, self).__init__(NphOrderDao())
+
+    def update_with_patch(self, id_, resource, expected_version):
+        pass
+
+    def put(self, nph_participant_id, rdr_order_id):
+        with database_factory.get_database().session() as session:
+            self.dao.set_order_cls(request.get_data())
+            order = self.dao.order_cls
+            self.dao.validate(rdr_order_id, nph_participant_id, session)
+            self.dao.order_sample_dao.update_order_sample(order, rdr_order_id, session)
+            self.dao.update_order(rdr_order_id, nph_participant_id, session)
+            session.commit()
+        return construct_response(order), 201
+
+    def post(self, nph_participant_id: str):
+        with database_factory.get_database().session() as session:
+            try:
+                self.dao.set_order_cls(request.get_data())
+                order = self.dao.order_cls
+                exist, time_point_id = self.dao.get_study_category_id(session)
+                if not exist:
```

Comment:
If I'm following this correctly `get_study_category_id` will try to find the timepoint's id for the order based on the names given for the module, visitType, and timepoint. And if they're not found, `insert_study_category_with_session` creates them all in the database.

So that works well if a new order comes in for `NPH Module 1`->`LMT`->`15 min` (all StudyCategories would be created). But then if we get another order for `NPH Module 1`->`LMT`->`30 min` they would all be created again, right? We want to be sure to add `30 min` to the `LMT` category we already have.